src/Forms/Shared/Samples/Map/OpenMobileMap/OpenMobileMap.xaml.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright 2017 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Esri.ArcGISRuntime.Mapping;
+using Xamarin.Forms;
+
+namespace ArcGISRuntimeXamarin.Samples.OpenMobileMap
+{
+    public partial class OpenMobileMap : ContentPage
+    {
+        // Hold a reference to a Mobile Map Package
+        private MobileMapPackage MyMapPackage;
+
+        public OpenMobileMap()
+        {
+            InitializeComponent();
+
+            Title = "Open mobile map (map package)";
+
+            // Create the UI, setup the control references and execute initialization
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            // Because each platform handles resource embedding differently,
```

Comment:
Wow, this section of code (next 20 lines or so) is brilliant! Well done using the conditional if/def statements. My only comment here that you add more comments of what is occurring in each platform. Are there possibly any Xamarin or Microsoft documents links that you could provide on the syntax differences for each platform for getting the assembly and folder in order to construct the path to the .mmpk?